corehq/apps/smsforms/models.py

```diff
@@ -111,13 +111,10 @@ def close(self):
         if not self.session_is_open:
             return
 
-        try:
-            if self.submit_partially_completed_forms:
-                submit_unfinished_form(self)
-        finally:
-            # this needs to be called after the submission, but regardless of whether it succeeded
-            # thus the try/finally
-            self.mark_completed(False)
+        if self.submit_partially_completed_forms:
+            submit_unfinished_form(self)
```

Comment:
Since this no longer marks completed if this fails, I think every place that calls `session.close()` will have to implement something like what you have below?

Reply:
I added in 4e19d22c5e2a0cb60c6345c9286e9eb165a3965b to make the retry behavior optional. The only other place outside of tests I see `close` called is in `close_all_open_sms_sessions`, which looks like it's used in places where we likely don't want to be waiting on retries (e.g., we're about to open a new session and are just making sure that any open sessions have been cleared).

However, it's still possible to hit errors when closing all sessions, and it does happen: https://sentry.io/organizations/dimagi/issues/1725304439/ I'll take a look at using `gevent` to make `close_all_open_sms_sessions` wait until all `close_session` tasks have returned - I'm not quite sure how gevent and retried tasks will work together.

Even pending that, I think this PR is an improvement over the existing behavior, do you agree?